clients/go/marshal.go

```diff
@@ -0,0 +1,37 @@
+package raccoon
+
+import (
+	"errors"
+
+	"encoding/json"
+
+	"google.golang.org/protobuf/proto"
+)
+
+type Message interface{}
+
+var (
+	// json clickstream marshaler
+	JSON = json.Marshal
+
+	// proto clickstream marshaler
+	PROTO = func(m interface{}) ([]byte, error) {
+		msg, ok := m.(proto.Message)
+		if !ok {
+			return nil, errors.New("unable to marshal non proto")
+		}
+		return proto.Marshal(msg)
+	}
+)
+
+// Marshaler defines a conversion for clickstream message to byte sequence.
+type Marshaler interface {
+	Marshal(any Message) ([]byte, error)
+}
+
+type MarshalFunc func(interface{}) ([]byte, error)
```

Comment:
Do we need this type along with Marshaler? Can this just one type `type Marshaler func(interface{}) ([]byte, error)`  which Send can simply execute?

Reply:
yes, we can do that, I just wanted to give the name `Marshal` to the `func`

Reply:
Updated

Reply:
updated and also renamed